### PR TITLE
Stories library from s3 as binary dependency

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR removes the `stories-android` git submodule and instead adds it as a binary dependency. It depends on the following PRs:

* https://github.com/Automattic/stories-android/pull/713 adds support for publishing `stories-android` artifacts to S3. It needs to be merged first, so the version that's used in this PR can be updated.
* https://github.com/wordpress-mobile/WordPress-Android/pull/15368 adds localization support for the `stories-android` binary dependency. **This also fixes the CI error, so please ignore it for this PR.**
* https://github.com/wordpress-mobile/WordPress-Android/pull/15369 cleans up the CircleCI steps from the submodule commands.

It also adds support for composite build for `stories-android` which works exactly the same way as the other libraries.

_Note that I am leaving this PR as a draft, because it depends on other PRs, but I'd appreciate a review so that it can all be merged at once. Thanks!_

**To test:**
* Smoke test the stories publishing flow
* Smoke test the stories publishing flow by adding stories-android as a composite build. You can find the instructions for this in the `local-builds.gradle-example` file:

1. `cp local-builds.gradle-example local-builds.gradle`
2. Remove the `//` prefix for `localStoriesAndroidPath` and make sure the relevant path is correct
3. Go into `stories-android` folder and make sure `publish-to-s3` branch is checked out
4. Re-sync Android Studio and test the app.

## Regression Notes
1. Potential unintended areas of impact
Stories-android related features might stop working

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing the stories-android feature

3. What automated tests I added (or what prevented me from doing so)
N/A

**PR submission checklist:**

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +0,0 @@
-[submodule "libs/stories-android"]
-	path = libs/stories-android
-	url = ../../Automattic/stories-android.git
-	shallow = true
```
Comment (mokagio) [inline .gitmodules:4]: 🎉 long time coming! Well done 👏

**Review by mokagio — APPROVED** [review]
Looking good. I tested this by building locally and publishing [a story](https://giotestdotsite.wordpress.com/2021/09/27/wpandroid-stories-via-s3/) with the library fetched via S3 and [another one](https://giotestdotsite.wordpress.com/2021/09/27/untitled/) with the composite build setup. Both worked fine.

**Comment by AliSoftware** [comment]
As discussed in Slack (p1633018037416900/1632506656.347400-slack-CC7L49W13), moving the milestone of those to 18.5, and we should be ready to merge these after today's code-freeze 🙂

**Review by mzorz — APPROVED** [review]
Tested this along with https://github.com/Automattic/stories-android/pull/713#pullrequestreview-771307682 everything looks good, thanks for these changes @oguzkocer ! 
:shipit:

File: .gitmodules (hunk 1 above)
Comment (AliSoftware) [reply .gitmodules:4]: Thanks again @oguzkocer for this indeed ❤️, this will be so much better with all submodules gone 🎉

**Pull request merged**